### PR TITLE
Modifying Dockerfile to build bittensor from repository version and not from github

### PR DESCRIPTION
**Pull request by eduardogr**
- Modifying Dockerfile to build pip package from local repository and not from github.
- Version to 3.5.1 in VERSION file and code.
- Adding circleci check for version update check in release and hotfix branches

After this PR, no version 3.5.2 will generated but a rebuilt of the 3.5.1 docker image will be done.

### CODE REVIEW

**Review by unconst — APPROVED** [review]

**Comment by unconst** [comment]
LGTM

File: .circleci/config.yml [hunk 1]
```diff
@@ -123,3 +135,11 @@ workflows:
...
           #requires:
             #- build-and-test
+
+  check-version-updated:
+    jobs:
+      - check-version-updated:
+          filters:
+            branches:
+              only:
+                - /^(release|hotfix)/.*/
```
Comment (Eugene-hu) [inline .circleci/config.yml:145]: So all branches with /release/ or /hotfix/ in its name will have to update their version?

**Review by Eugene-hu — COMMENTED** [review]
LGTM

**Review by Eugene-hu — APPROVED** [review]

File: .circleci/config.yml (hunk 1 above)
Comment (eduardogr) [reply .circleci/config.yml:145]: yes, that's the idea.
branches like:
  - release/*
  - hotfix/*
  
will add all we need to release, so when branches arrive the default branch we can trigger the release

Comment (eduardogr) [reply .circleci/config.yml:145]: this is to ensure that we are going to have the version update in files when release without need to push to master (we do not want that) or a new PR

**Review by camfairchild — APPROVED** [review]
LGTM

**Pull request merged**